oidc.py

```diff
@@ -1,26 +1,112 @@
+import json
+import requests
+import time
+
+from django.views.generic import View
+from rest_framework_simplejwt.tokens import RefreshToken
 from mozilla_django_oidc.auth import OIDCAuthenticationBackend
+from authlib.jose import jwt
 from django.conf import settings
-from django.shortcuts import render
+from django.utils.decorators import method_decorator
+from django.views.decorators.csrf import csrf_exempt
+from django.http import JsonResponse
+from django.contrib.auth import get_user_model
+
+UserModel = get_user_model()
 
 def status(request):
-    return {
-        "OIDC_ENABLED": settings.OIDC_ENABLED,
-    }
+    return {"OIDC_ENABLED": settings.OIDC_ENABLED}
+
 
-def updateUser(user, claims):
+def _updateUser(user, claims):
     user.email = claims.get("email")
     user.first_name = claims.get("given_name")
     user.last_name = claims.get("family_name")
     return user
 
+
+def _read_private_key(keyfile):
+    with open(keyfile, "r") as f:
```

Comment:
maybe add a `try` and catch the case where the env variable `OIDC_PRIVATE_KEYFILE` is not set, with an explicit error

Reply:
Now settings throws an exception if the OIDC is enabled, but misses some configuration parameters